src/Components/BranchList/index.tsx

```diff
@@ -42,40 +42,48 @@ function BranchTree(branches: any) {
     )
 }
 
-export default class BranchList extends Component<any, {branches: ReturnType<typeof getBranchTree>, head?: BranchObj}> {
+type Props = {
+    branches?: BranchesObj
```

Comment:
Couldnt this be an array of branches instead? The obj-key and the Type seems mismatched.

Reply:
It's an object with arrays of BranchObj. The type `BranchesObj` could be renamed to something better representing the actual type. Don't think that's an issue that should be resolved in this PR though.